app/controllers/versions_controller.rb

```diff
@@ -116,13 +116,13 @@ def destroy
       flash[:error] = call.errors.full_messages
     end
 
-    redirect_to settings_project_path(tab: 'versions', id: @project)
+    redirect_to settings_project_path(@project, tab: 'versions')
   end
 
   private
 
   def redirect_back_or_version_settings
-    redirect_back_or_default(settings_project_path(tab: 'versions', id: @project))
+    redirect_back_or_default(settings_project_path(@project, tab: 'versions'))
```

Comment:
Why not use the routes you defined, e.g. `settings_versions_project_path(@project)`?